src/v/kafka/server/handlers/topics/validators.h

```diff
@@ -411,6 +412,41 @@ struct write_caching_configs_validator {
     }
 };
 
+struct delete_retention_ms_validator {
+    static constexpr const char* error_message
+      = "Unsupported delete.retention.ms configuration, cannot be enabled "
+        "at the same time as redpanda.remote.read or redpanda.remote.write.";
+    static constexpr const auto config_name
+      = topic_property_delete_retention_ms;
+    static constexpr error_code ec = error_code::invalid_config;
+
+    static bool is_valid(const creatable_topic& c) {
+        const auto config_entries = config_map(c.configs);
+        auto end = config_entries.end();
+        bool delete_retention_ms
+          = (config_entries.find(topic_property_delete_retention_ms) != end);
+
+        try {
+            auto shadow_indexing_mode = get_shadow_indexing_mode(
+              config_entries);
+            // Cannot set delete_retention_ms at the same time as any tiered
+            // storage properties.
+            if (
+              delete_retention_ms
+              && shadow_indexing_mode
+                   != model::shadow_indexing_mode::disabled) {
+                return false;
+            }
+        } catch (const boost::bad_lexical_cast&) {
+            // Caught a bad configuration exception.
+            // Return true for now- this will error out in a later stage.
```

Comment:
Mentioned in the walkthrough, not for this PR, but we should consider reordering validation so each validator don't need to do these checks, if possible

Reply:
Have this as a note for future work in a separate PR.